### PR TITLE
Fix/deploy singleton

### PR DESCRIPTION
**Pull request by dvilelaf**
I'm deploying the contract called Singleton.sol just in case, as all proxy creations seems to use its address. I think the proper one is the main contract GnosisSafeL2 as I was doing until now, but that one (despite seeming to be working) doesn't let us call masterCopy() (I'm using that to check if it works) and looking at the tests it definitely looks like it's not deployed. 

The singleton.sol contract just stores an address, so it's unlikely that's what we need. Furthermore, creating a proxy with its address end in a `Transactioransaction reverted without a reason string` error, the same as shows in this test [here](https://github.com/gnosis/safe-contracts/blob/main/test/factory/ProxyFactory.spec.ts#L110-L111) when they use the wrong init data.

### CODE REVIEW

File: deploy/6_proxy_safe.js [hunk 1]
```diff
@@ -32,17 +36,21 @@ module.exports = async (hre) => {
     const safe_contracts = require("@gnosis.pm/safe-contracts");
     proxy_address = await safe_contracts.calculateProxyAddress(proxy_factory_contract, master_contract.address, setupData, nonce);
 
-    await proxy_factory_contract.createProxyWithNonce(master_contract.address, setupData, nonce).then((tx) => tx.wait());
+    await proxy_factory_contract.createProxyWithNonce(singleton_contract.address, setupData, nonce).then((tx) => tx.wait());
```
Comment (dvilelaf) [inline deploy/6_proxy_safe.js:39]: I don't think this is correct.

Comment (dvilelaf) [reply deploy/6_proxy_safe.js:39]: This is the createProxyWithNonce tx data when called with master_contract address:
```
{
  to: '0xDc64a140Aa3E981100a9becA4E685f962f0cF6C9',
  from: '0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266',
  contractAddress: null,
  transactionIndex: 0,
  gasUsed: BigNumber { _hex: '0x04a91f', _isBigNumber: true },
  logsBloom: '0x00000000000000000000000000000000000000000000000000000008000000000000000000000000400000000000000000000000000000000000000000000000000010000000000000000000000000000000000000000009000000000000000000000000000000000500000000000000000000000000000000020000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000080000000000000000000000000000000000000000000000000000000000000000000000000000000020000004000000000000000000080000000000000000000000000020000001000000000100000000000000',
  blockHash: '0x5bce43bf6b499965e6c61faf61a20f8f8b23f88b4c56f4ff90a33b5a127911d4',
  transactionHash: '0x316b33c681945617f7ef7dfa06cb4f8c562babe003c7c245e72b775dd7dfb7b9',
  logs: [
    {
      transactionIndex: 0,
      blockNumber: 78,
      transactionHash: '0x316b33c681945617f7ef7dfa06cb4f8c562babe003c7c245e72b775dd7dfb7b9',
      address: '0xd29d410Bb17B3c67B7E0Cd2C9325c6d82B6889fb',
      topics: [Array],
      data: '0x00000000000000000000000000000000000000000000000000000000000000800000000000000000000000000000000000000000000000000000000000000003000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000004000000000000000000000000bcd4042de499d14e55001ccbb24a551f3b95409600000000000000000000000071be63f3384f5fb98995898a86b02fb2426c5788000000000000000000000000fabb0ac9d68b0b445fb7357272ff202c5651694a0000000000000000000000001cbd3b2770909d4e10f157cabc84c7264073c9ec',
      logIndex: 0,
      blockHash: '0x5bce43bf6b499965e6c61faf61a20f8f8b23f88b4c56f4ff90a33b5a127911d4'
    },
    {
      transactionIndex: 0,
      blockNumber: 78,
      transactionHash: '0x316b33c681945617f7ef7dfa06cb4f8c562babe003c7c245e72b775dd7dfb7b9',
      address: '0xDc64a140Aa3E981100a9becA4E685f962f0cF6C9',
      topics: [Array],
      data: '0x000000000000000000000000d29d410bb17b3c67b7e0cd2c9325c6d82b6889fb000000000000000000000000e7f1725e7734ce288f8367e1bb143e90bb3f0512',
      logIndex: 1,
      blockHash: '0x5bce43bf6b499965e6c61faf61a20f8f8b23f88b4c56f4ff90a33b5a127911d4'
    }
  ],
  blockNumber: 78,
  confirmations: 1,
  cumulativeGasUsed: BigNumber { _hex: '0x04a91f', _isBigNumber: true },
  effectiveGasPrice: BigNumber { _hex: '0x3b9b5ea0', _isBigNumber: true },
  status: 1,
  type: 2,
  byzantium: true,
  events: [
    {
      transactionIndex: 0,
      blockNumber: 78,
      transactionHash: '0x316b33c681945617f7ef7dfa06cb4f8c562babe003c7c245e72b775dd7dfb7b9',
      address: '0xd29d410Bb17B3c67B7E0Cd2C9325c6d82B6889fb',
      topics: [Array],
      data: '0x00000000000000000000000000000000000000000000000000000000000000800000000000000000000000000000000000000000000000000000000000000003000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000004000000000000000000000000bcd4042de499d14e55001ccbb24a551f3b95409600000000000000000000000071be63f3384f5fb98995898a86b02fb2426c5788000000000000000000000000fabb0ac9d68b0b445fb7357272ff202c5651694a0000000000000000000000001cbd3b2770909d4e10f157cabc84c7264073c9ec',
      logIndex: 0,
      blockHash: '0x5bce43bf6b499965e6c61faf61a20f8f8b23f88b4c56f4ff90a33b5a127911d4',
      removeListener: [Function (anonymous)],
      getBlock: [Function (anonymous)],
      getTransaction: [Function (anonymous)],
      getTransactionReceipt: [Function (anonymous)]
    },
    {
      transactionIndex: 0,
      blockNumber: 78,
      transactionHash: '0x316b33c681945617f7ef7dfa06cb4f8c562babe003c7c245e72b775dd7dfb7b9',
      address: '0xDc64a140Aa3E981100a9becA4E685f962f0cF6C9',
      topics: [Array],
      data: '0x000000000000000000000000d29d410bb17b3c67b7e0cd2c9325c6d82b6889fb000000000000000000000000e7f1725e7734ce288f8367e1bb143e90bb3f0512',
      logIndex: 1,
      blockHash: '0x5bce43bf6b499965e6c61faf61a20f8f8b23f88b4c56f4ff90a33b5a127911d4',
      args: [Array],
      decode: [Function (anonymous)],
      event: 'ProxyCreation',
      eventSignature: 'ProxyCreation(address,address)',
      removeListener: [Function (anonymous)],
      getBlock: [Function (anonymous)],
      getTransaction: [Function (anonymous)],
      getTransactionReceipt: [Function (anonymous)]
    }
  ]
}
```

Comment (dvilelaf) [reply deploy/6_proxy_safe.js:39]: The deployment calculated address is `0xd29d410Bb17B3c67B7E0Cd2C9325c6d82B6889fb`. It also appears on the `ProxyCreation` event data.

**Comment by DavidMinarsch** [comment]
Why not simply follow what they do? https://github.com/gnosis/safe-contracts/tree/main/src/deploy

File: deploy/6_proxy_safe.js [hunk 2]
```diff
@@ -32,17 +36,21 @@ module.exports = async (hre) => {
...
     for (let i = 10; i < 14; i++) {
-        path = path_string.replace('n', i.toString())
+        path = path_string.replace('n', i.toString());
         wallet = ethers.Wallet.fromMnemonic(default_mnemonic, path);
         console.log("Safe proxy owner", (i - 10).toString(), "address:", wallet.address, "key:", wallet.privateKey);
     }
+
+    // Verify proxy deployment
+    proxy_contract = await hre.ethers.getContractAt("GnosisSafeProxy", proxy_address);
+    console.log(proxy_contract.masterCopy());
```
Comment (dvilelaf) [inline deploy/6_proxy_safe.js:55]: Is this correct? Andrey pointed out that we should test with `masterCopy()` and try to retrieve the safe address that way. This returns a `TypeError: proxy_contract.masterCopy is not a function` ATM. That is defined in the proxy contract as:
```
interface IProxy {
    function masterCopy() external view returns (address);
}
```

Comment (DavidMinarsch) [reply deploy/6_proxy_safe.js]: I finished it now, was just a small fix.

**Review by DavidMinarsch — APPROVED** [review]
LGTM - now we are finally fully set up :)

**Pull request merged**